lib/bdev/malloc/blockdev_malloc.c

```diff
@@ -126,7 +126,7 @@ blockdev_malloc_read(struct malloc_disk *mdisk, struct copy_task *copy_req,
 	SPDK_TRACELOG(SPDK_TRACE_MALLOC, "read %lu bytes from offset %#lx to %p\n",
 		      nbytes, offset, buf);
 
-	return spdk_copy_submit(copy_req, buf, mdisk->malloc_buf + offset,
+	return spdk_copy_submit(copy_req, buf, (void *)((char *)mdisk->malloc_buf + offset),
```

Comment:
This is intentionally using the GCC void pointer arithmetic extension: https://gcc.gnu.org/onlinedocs/gcc/Pointer-Arith.html

The same pattern is used throughout the code, so I don't think changing the cases in just this one file is a good idea (and it makes the code harder to read).